src/Compilers/Core/Portable/ConstantValueSpecialized.cs

```diff
@@ -201,6 +210,21 @@ internal override string GetValueToDisplay()
             {
                 return (_value == null) ? "null" : string.Format("\"{0}\"", _value);
             }
+
+            public override string ToString(string? format, IFormatProvider? provider)
+            {
+                int formatLength = RopeValue.Length;
+                if (format is not null && int.TryParse(format, out var len) &&
+                    len >= 0)
+                {
+                    formatLength = len;
+                }
+
+                bool trim = formatLength >= 3 && formatLength < RopeValue.Length;
+                var formatString = @"""{0}""" + (trim ? "..." : string.Empty);
+
+                return string.Format(formatString, trim ? RopeValue.ToString(formatLength - 3) : RopeValue);
```

Comment:
>trim ? [](http://example.com/codeflow?start=51&length=6)

Minor: Would it be simpler to have separate `string.Format()` calls for the two cases?
```
return (formatLength >= 3 && formatLength < RopeValue.Length) ?
    string.Format(""{0}...", RopeValue.ToString(formatLength - 3)) :
    string.Format(""{0}"", RopeValue);
```